tests/test_scikit_learn_checks.py

```diff
@@ -0,0 +1,79 @@
+"""Tests using Scikit-Learn's bundled estimator_checks."""
+
+from distutils.version import LooseVersion
+
+import pytest
+
+from sklearn import __version__ as sklearn_version
+from sklearn.datasets import load_iris
+from sklearn.utils.estimator_checks import check_no_attributes_set_in_init
+
+from scikeras.wrappers import KerasClassifier
+from scikeras.wrappers import KerasRegressor
+
+from .mlp_models import dynamic_classifier
+from .mlp_models import dynamic_regressor
+from .testing_utils import basic_checks
+from .testing_utils import parametrize_with_checks
+
+
+@parametrize_with_checks(
+    estimators=[
+        KerasClassifier(
+            build_fn=dynamic_classifier,
+            # Set batch size to a large number (larger than X.shape[0] is the goal)
+            # if batch_size < X.shape[0], results will very
+            # slightly if X is shuffled.
+            # This is only required for this tests and is not really
+            # applicable to real world datasets
+            batch_size=1000,
+            optimizer="adam",
+        ),
+        KerasRegressor(
+            build_fn=dynamic_regressor,
+            # Set batch size to a large number (larger than X.shape[0] is the goal)
+            # if batch_size < X.shape[0], results will very
+            # slightly if X is shuffled.
+            # This is only required for this tests and is not really
+            # applicable to real world datasets
+            batch_size=1000,
+            optimizer="adam",
+            loss=KerasRegressor.root_mean_squared_error,
+        ),
+    ],
+    ids=["KerasClassifier", "KerasRegressor"],
+)
+def test_fully_compliant_estimators(estimator, check):
```

Comment:
@stsievert I know we went back and forth on checking that non-subclassed models can pass all tests. Here is proof that they do. As you suggested, the only test they fail is `check_no_attributes_set_in_init`. I added a separate tests specifically for that below.